tests/consensus_tests/test_resharding.py

```diff
@@ -2,991 +2,307 @@
-    }
+    # Remove target peer
+    resp = requests.delete(f"{peer_uris[0]}/cluster/peer/{peer_ids[-1]}?force=true")
+    assert_http_ok(resp)
 
-    peer_api_uris, _peer_dirs, _bootstrap_uri = start_cluster(tmp_path, 3, None, extra_env=env)
+    # Wait for resharding to abort
+    wait_for_collection_resharding_operations_count(peer_uris[0], COLLECTION_NAME, 0)
 
-    # Create collection, insert points
-    create_collection(peer_api_uris[0], shard_number=1, replication_factor=3)
-    wait_collection_exists_and_active_on_all_peers(
-        collection_name=COLLECTION_NAME,
-        peer_api_uris=peer_api_uris,
-    )
-    upsert_random_points(peer_api_uris[0], num_points)
+def test_resharding_try_remove_target_shard(tmp_path: pathlib.Path):
+    # Bootstrap resharding cluster
+    peer_uris, peer_ids = bootstrap_resharding(tmp_path)
 
-    sleep(1)
+    # Try to remove target shard
+    resp = requests.post(f"{peer_uris[0]}/collections/{COLLECTION_NAME}/cluster", json={
+        "drop_replica": {
+            "peer_id": peer_ids[0],
+            "shard_id": 3,
+        }
+    })
 
-    # Assert node shard and point sum count
-    for uri in peer_api_uris:
-        assert check_collection_local_shards_count(uri, COLLECTION_NAME, 1)
-        assert check_collection_local_shards_point_count(uri, COLLECTION_NAME, num_points)
-
-    # Reshard 3 times in sequence
-    for shard_count in range(2, 5):
-        # Start resharding
-        r = requests.post(
-            f"{peer_api_uris[0]}/collections/{COLLECTION_NAME}/cluster", json={
-                "start_resharding": {
-                    "direction": "up"
-                }
-            })
-        assert_http_ok(r)
-
-        # Wait for resharding to start
-        wait_for_collection_resharding_operations_count(peer_api_uris[0], COLLECTION_NAME, 1)
-
-        # Continuously assert point count on all peers, must be stable
-        # Stop once all peers reported completed resharding
-        while True:
-            for uri in peer_api_uris:
-                assert get_collection_point_count(uri, COLLECTION_NAME, exact=True) == num_points
-                cardinality_count = get_collection_point_count(uri, COLLECTION_NAME, exact=False)
-                assert cardinality_count >= num_points / 2 and cardinality_count < num_points * 2
-
-            all_completed = True
-            for uri in peer_api_uris:
-                if not check_collection_resharding_operations_count(uri, COLLECTION_NAME, 0):
-                    all_completed = False
-                    break
-            if all_completed:
-                break
-
-        # Assert node shard count
-        for uri in peer_api_uris:
-            assert check_collection_local_shards_count(uri, COLLECTION_NAME, shard_count)
+    assert_http(resp, 400)
 
-    sleep(1)
+@pytest.mark.parametrize("direction, peers", [("up", 3), ("down", 3)])
+def test_resharding_forward(tmp_path: pathlib.Path, direction: Literal["up", "down"], peers: int):
```

Comment:
Maybe we can also parameterize the number of shards/replicas (if that isn't too expensive)